dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/PreCheckSecurityOwnershipValidationHookTest.java

```diff
@@ -484,7 +593,63 @@ void verifyValidationFailsForEnrollmentUsingDeleteStrategyAndUserWithoutCascadeA
         validatorToTest.validateEnrollment( reporter, enrollment );
 
         assertTrue( reporter.hasErrors() );
-        assertThat( reporter.getReportList().get( 0 ).getErrorCode(), is( E1103 ) );
+        assertTrue( reporter.hasErrorReport( err -> E1103.equals( err.getErrorCode() ) &&
+            TrackerType.ENROLLMENT.equals( err.getTrackerType() ) &&
+            enrollment.getUid().equals( err.getUid() ) ) );
+    }
+
+    @Test
+    void verifyValidationFailsForEnrollmentDeletionAndUserWithoutProgramWriteAccess()
+    {
+        String enrollmentUid = CodeGenerator.generateUid();
+        Enrollment enrollment = Enrollment.builder()
+            .enrollment( enrollmentUid )
+            .orgUnit( ORG_UNIT_ID )
+            .trackedEntity( TEI_ID )
+            .program( PROGRAM_ID )
+            .build();
+        when( ctx.getStrategy( enrollment ) ).thenReturn( TrackerImportStrategy.DELETE );
+        when( ctx.getProgram( PROGRAM_ID ) ).thenReturn( program );
+        when( ctx.getOrganisationUnit( ORG_UNIT_ID ) ).thenReturn( organisationUnit );
+        when( aclService.canDataWrite( user, program ) ).thenReturn( false );
+        when( aclService.canDataRead( user, program.getTrackedEntityType() ) ).thenReturn( true );
+        when( organisationUnitService.isInUserHierarchyCached( user, organisationUnit ) )
+            .thenReturn( true );
+        reporter = new ValidationErrorReporter( ctx, enrollment );
+
+        validatorToTest.validateEnrollment( reporter, enrollment );
+
+        assertTrue( reporter.hasErrors() );
+        assertTrue( reporter.hasErrorReport( err -> E1091.equals( err.getErrorCode() ) &&
+            TrackerType.ENROLLMENT.equals( err.getTrackerType() ) &&
+            enrollment.getUid().equals( err.getUid() ) ) );
+    }
```

Comment:
Feels very repetitive around here. Really should get some helpers to avoid duplicating setup and asserts.

Reply:
I will tackle this later on